pkg/clusteragent/admission/mutate/config/mutator.go

```diff
@@ -125,13 +117,18 @@ func (i *Mutator) MutatePod(pod *corev1.Pod, _ string, _ dynamic.Interface) (boo
 	}
 
 	// Inject DD_AGENT_HOST
-	switch injectionMode(pod, i.config.mode) {
+	switch injectionMode(pod, i.config.mode, i.config.csiEnabled) {
 	case hostIP:
 		injectedConfig = mutatecommon.InjectEnv(pod, agentHostIPEnvVar)
 	case service:
 		injectedConfig = mutatecommon.InjectEnv(pod, agentHostServiceEnvVar)
 	case socket:
-		injectedVolumes := i.injectSocketVolumes(pod)
+		injectedVolumes := i.injectSocketVolumes(pod, false)
```

Comment:
I think the four lines are the same except the param to `injectSocketVolumes`, so this could be simplified to something like:
```go
case socket, csi:
	useCSI := (mode == csi)
	injectedVolumes := i.injectSocketVolumes(pod, useCSI)
	injectedEnv := mutatecommon.InjectEnv(pod, traceURLSocketEnvVar)
	injectedEnv = mutatecommon.InjectEnv(pod, dogstatsdURLSocketEnvVar) || injectedEnv
	injectedConfig = injectedVolumes || injectedEnv
```